build.sbt

```diff
@@ -258,12 +258,12 @@ lazy val framelessSettings = Seq(
   },
 ) ++ consoleSettings
 
-lazy val spark30Settings = Seq(
+lazy val spark31Settings = Seq(
   crossScalaVersions := Seq(Scala212)
 )
 
-lazy val spark31Settings = Seq(
-  crossScalaVersions := Seq(Scala212)
+lazy val spark32Settings = Seq(
+  mimaPreviousArtifacts := Set.empty
 )
```

Comment:
As a rule-of-thumb, you should never set `mimaPreviousArtifacts := Set.empty`, this is easy to forget and break binary-compatibiliy in a future release by mistake.

There are basically two situations:
1. You are not publishing the module, so use `.enablePlugins(NoPublishPlugin)` to disable MiMa.
2. It is a new module that you have not published yet. So use this setting to indicate which version it will be introduced, and MiMa will automatically start from that version. 
    ```scala
     tlVersionIntroduced := Map("2.12" -> "0.13.0", "2.13" -> "0.13.0")
    ```

Reply:
@armanbilge thanks! is this intel documented anywhere? What will be the setting for the artifact I publish but it has not been published before? (To be clear: 3.2 is the new artifact and we don't need to compare it with the previous version)

Reply:
> is this intel documented anywhere?

Sort of. This particular setting is documented here:
https://typelevel.org/sbt-typelevel/faq.html#how-do-i-indicate-the-first-version-that-i-published-scala-3-artifacts-for

Although perhaps it can use some other examples like for your situation.

> What will be the setting for the artifact we publish (first time) and it has not been published before?

What is the first version you plan to publish it in?

Reply:
@armanbilge o i c, 0.13 is the first version, so we need to add `tlVersionIntroduced := Map("2.12" -> "0.13.0", "2.13" -> "0.13.0")` as an indication that the first artifact was published as the 0.13 version for scala 2.12 and 2.13. 

Makes sense now, thanks!

Reply:
Right, exactly, you got it! :) And this way you don't need to remember to change it again after publishing 0.13.0, MiMa will start working automatically.